src/common/checksums.h

```diff
@@ -163,7 +163,7 @@ class OCSYNC_EXPORT ValidateChecksumHeader : public QObject
 
 signals:
     void validated(const QByteArray &checksumType, const QByteArray &checksum);
-    void validationFailed(const QString &errMsg);
+    void validationFailed(const QString &errMsg, const QByteArray &checksumType, const QByteArray &checksum, const QString &filePath);
```

Comment:
Why not providing default values to the new arguments? That would require less changes.